app/routes/admin/posts.py

```diff
@@ -153,6 +157,26 @@ def edit(post_id):
                            upload_form=upload_form)
 
 
+@posts.route('/posts/preview/<slug>')
+def preview(slug):
+    if BlogPost.objects(slug=slug).count() != 1:
+        abort(404)
+    post = BlogPost.objects().get(slug=slug)
+    if not post.date_published:
+        post.date_published = datetime.now()
```

Comment:
Well it's not actually being published. It's just that if the post isn't published than post.date_published is null, causing the preview page to crash (since it displays the date published). So I thought it needed a place holder of some sort, so I set date published to the current time. But that date doesn't get saved in the database.